plugins/plugin-urlfactory/src/main/java/org/eclipse/che/plugin/urlfactory/URLFetcher.java

```diff
@@ -89,4 +94,22 @@ public String fetch(@NotNull URLConnection urlConnection) {
   protected long getLimit() {
     return MAXIMUM_READ_BYTES;
   }
+
+  /**
+   * Simple method to sanitize the Git urls like &quot;https://github.com/demo.git&quot; or
+   * &quot;http://myowngit.example.com/demo.git&quot;
+   *
+   * @param url - the String format of the url
+   * @return if the url ends with .git will return the url without .git otherwise return the url as
+   *     it is
+   */
+  String sanitized(String url) {
```

Comment:
When can it happen?

Reply:
when i do something like http://<my-che>/f?url=http://github.com/demo/demo.git to create a workspace from factory. As users has 99% chances of using this url `http://github.com/demo/demo.git` as they usually copy the url from github which includes .git. WDYT?

Reply:
I don't understand this use-case. When do we need to sanitized url?

Reply:
Try doing a workspace creation using https://codenvy.io/f?url=https://github.com/eclipse/che, this will work as expected and load the right docker image suited for che development (Assuming it’s differnt from default Ubuntu JDK image). But try the same with https://codenvy.io/f?url=https://github.com/eclipse/che.git, note .git at the end of url parameter the factory creation fails as it will not be able to load .factory.json from the url https://codenvy.io/f?url=https://github.com/eclipse/che.git

Reply:
@skabashnyuk it happens when creating a Factory with a git url as a query param.

Reply:
@skabashnyuk for the url tests if you use my repo `https://github.com/skshiftio/mydemo` which has a factory JSON in it, actually this should create a workspace with Spring boot image but if you use url parameter like `https://github.com/skshiftio/mydemo.git` then it will load the ubuntu_jdk8 image as it is not able to read the .factory.json.